### ISSUE TITLE
Can't open personal private folder - reset message included

### ISSUE THREAD

**Issue by thanandorn** [keybase/client#8794]
Please reset my access to the private folder

`$ keybase sign -m "Please reset folder /keybase/private/thanandorn because all the devices that have access have been revoked."`

```
BEGIN KEYBASE SALTPACK SIGNED MESSAGE. kXR7VktZdyH7rvq v5weRa0zkS1DxuS VoVE2HRXmNQtJM5 QtJmhLoo5oATFzV 29pnt1ejBbkOxTm m1LisDisj2L27W0 LfPN0sX1u7Q7KdC hA1TZukHEePXrPa cq06qhX85bJxw0N juRNWDokUINcPWY JuQ3RabUcOWZxNH kzogaKdaj8Aa3jW pgNvWKl45e1gWTA 3rSmplM4Vgm6JfH AWqglO3sonCHNAw L7a3OgHdvLd8hP4 qRFw1DBPVgQfoHe 0yRNyluCodiGQTt nzfMh0qS014jVuR xRJTTFkdqvWronJ 71612n4UDnXIpih sYRbnVaJ9IyAsUY . END KEYBASE SALTPACK SIGNED MESSAGE.
```

**Comment by strib** [comment]
@thanandorn: I can do that, but can you please include the current time and date in the signed-over message?  Something like this, substituting the time and date where indicated:

```
keybase sign -m "<DATE_AND_TIME>: Please reset folder /keybase/private/thanandorn because all the devices that have access have been revoked."
```

Thanks!

**Comment by thanandorn** [comment]
@strib Here you are. Hope I've got it right this time.

`keybase sign -m "Oct 4, 2017 10:16:34: Please reset folder /keybase/private/thanandorn because all the devices that have access have been revoked."`

```
BEGIN KEYBASE SALTPACK SIGNED MESSAGE. kXR7VktZdyH7rvq v5weRa0zkS1DxuS VoVE2HRXmNQtJM5 QtJmhLoo5oATFzV 29pnt1ejBcEBszD 8hWkxMKyDiGZqyJ BZcMmTQL44iLsrQ 6pSnL2qkbbKy9Lq WyLF4g6k4mbSsvr jXnSCyurEtS9F7y 8BEijArUu7WlxIg 1FJZp3fMchbMKYt Ts9SB9fO8iDGVkb 2M1uWtoct0SBgih HjbY8J4OQO97AIo XwSyMjs05DruYA7 4w8Jw8OlGxs29KU pj7S4RdVknYs3Jj IHIbApC4Xc8a5Zq naL2AicnPX6jv0b YNo2Lvhn5JaNDLB wX7MWE0Po1DGxzH i6aSrK3SAItsrF9 t1GFjoZG7lhFRm. END KEYBASE SALTPACK SIGNED MESSAGE.
```
Many thanks!

**Comment by strib** [comment]
I reset the folder.  Please check it out and close this if everything seems ok.  (Might require a Keybase restart.)

**Comment by thanandorn** [comment]
@strib Works now. Thank you very much.

**Comment by strib** [comment]
@thanandorn your private folder is still readable by the following, unrevoked devices on your account:

* `puppy review` (paper key)
* `mimix`
* `marcos`
* `sriracha`

Those seem to be the _only_ devices currently associated with your account, so the problem must be something else.  Can you please run `keybase log send`, and post the resulting log ID here, and I can look more into it?

**Comment by thanandorn** [comment]
@strib Here you are. Thx

```
------------
Success! Your log ID is:

  f1f673d96a17d9b2fafaa91c
```

**Comment by strib** [comment]
There was a regression in the linux builds around reading folders that had been written by old revoked devices, as in your case.  The latest linux release has the fix, so please try upgrading and close this out if things seem ok now.

**Comment by thanandorn** [comment]
@strib sorry its still not working after upgrading the client.

```
$ rpm -qi keybase
Name        : keybase
Version     : 2.0.0.20180530173728.f81992c7c9
Release     : 1
Architecture: x86_64
Install Date: Wed 30 May 2018 19:27:45 BST
Group       : Unspecified
Size        : 254090408
License     : BSD
Signature   : RSA/SHA256, Wed 30 May 2018 18:47:51 BST, Key ID 47484e50656d16c7
Source RPM  : keybase-2.0.0.20180530173728.f81992c7c9-1.src.rpm
Build Date  : Wed 30 May 2018 18:47:12 BST
Build Host  : cafc96653d13
Relocations : (not relocatable)
Summary     : Keybase command line client
Description :
Keybase command line client

$ ls -al /keybase/private/thanandorn/
ls: cannot open directory '/keybase/private/thanandorn/': Permission denied

$ ls -al /keybase/public/thanandorn/
total 2
-rw-------. 1 tron root 395 Aug 24  2016 id_rsa.pub
-rw-------. 1 tron root  22 Jan 26  2017 index.md
drwx------. 1 tron root 489 Apr  4  2017 tara
-rw-------. 1 tron root 370 Sep 20  2017 Vagrantfile
```

**Comment by strib** [comment]
Can you please do another `keybase log send` then?

**Comment by thanandorn** [comment]
```
Success! Your log ID is:

  09ac3cc6e14e89b7462f111c

```

Thank you.

**Comment by strib** [comment]
Thanks.  I've tracked down the issue.  Your client is actually still trying to access the old version of the folder that I reset back in October.  This is because of a complicated chain of events relating to a new chat feature, which I won't bother to detail here unless you're really interested.

Suffice it to say, this is a bug in our code, and I'm working with a few people to figure out a safe resolution for you.  Sorry for the inconvenience, and thanks for your patience.

(P.S.: There's no data loss or anything, the new folder is still there with all its data, we just need to fix things so that your client sees it instead of the older one.)

**Comment by thanandorn** [comment]
Summary of the problem would be interesting to see but I'll let you work on it first. Please take your time. Many thanks.

**Comment by strib** [comment]
@thanandorn sorry about the delay.  We reset some data on our servers that should let you access your new folder again.  First, run `keybase db nuke`.  Then, just to be safe, restart Keybase.  Then try again and let us know what happens. Thanks!

(I'll give a rundown of the problem in a separate message so it doesn't delay this one.)

**Comment by strib** [comment]
In the beginning, there was the file system (KBFS).  KBFS was built with a special-purpose keying mechanism for its data, which had some operational drawbacks.  (Like, if you add a new device, the rekeying isn't instantaneous; there is an asynchronous operation that has to happen and can fail silently.)

Then we introduced chat, which originally relied on the same keys that KBFS generates.  So under the hood, each chat was "powered" by a matching KBFS folder in terms of the keys it used.

Then we introduced teams, and at the same time we added a more general keying mechanism that doesn't have the drawbacks of the KBFS one. Adding a new device atomically gives that device access to all the keys it needs.  Our intent was (and is) to migrate all chats and KBFS folder to this new mechanisms, so they can all benefit.

Chat migration happened first (very recently) because it's a bit less complicated.  KBFS migration is almost ready, and in fact existing clients can already understand the migrated KBFS folders, when that eventually happens.  These migrations use the new keying mechanism by relying on "implicit teams" -- that is, in order to use the new team keying mechanism, we turn EVERYTHING into a team (such as your private folder and self-conversation) under the hood.  Every team has its own "sigchain", which is a the sequence of operations, signed by valid users of the team, to make sure Keybase isn't cheating anywhere.

For KBFS migration, we are going to store the folder ID in the sigchain itself (rather than as a separate mapping between folder name and ID in the KBFS server db, as it is today).  Chat also needs to store the ID in the sigchain, because it needs to be able to find the old KBFS keys for migrated chats.  So, as part of chat migration, it stored the folder ID for every existing chat in the sigchain for that chat.

In your case, because you had this old folder ID (that has previously been reset) lying around in our database, the chat migration code accidentally picked up the old ID instead of the new ID and put it in your sigchain.  Then, your KBFS folder saw that and started using its post-migration code, and treated that as the real folder ID instead of checking with the server.  So that's why it's not seeing the new, post-reset folder.

This is a bug, but it only affects users who've had a manual reset done to their folders AND had a chat conversation that matched that folder, AND got unlucky and the wrong folder ID was put into the sigchain.  We (hopefully) fixed it for you by just nuking the implicit team associated with your private folder/chat, and when you access it again a brand new implicit team will be made instead.
